.gitlab-ci.yml

```
@@ -28,27 +28,41 @@ variables:
   S3_BUCKET_NAME: "mender"
   S3_BUCKET_PATH: "mender-artifact"
   DOCKER_BUILDKIT: 1
+  DOCKER_PLATFORMS:
+    value: "linux/amd64,linux/arm64"
+    description: "Platforms for building and distributing container images"
 
 test:check-license-source:
   variables:
     LICENSE_HEADERS_IGNORE_FILES_REGEXP: './artifact/keyfactor/\(signer\|signer_test\)\.go'
 
 build:docker:
-  image: docker
+  image: docker:26-cli
   needs: []
   services:
-    - docker:dind
+    - docker:26-dind
   stage: build
+  before_script:
+    - docker login --username "$CI_REGISTRY_USER" --password "$CI_REGISTRY_PASSWORD" "$CI_REGISTRY"
+    - docker buildx create --platform "$DOCKER_PLATFORMS" --name builder --bootstrap --use
   script:
-    - docker build -t $DOCKER_REPOSITORY:pr .
-    - docker save $DOCKER_REPOSITORY:pr > image.tar
-  artifacts:
-    expire_in: 2w
-    paths:
-      - image.tar
+    - |
+      docker buildx build --push \
+        --platform "$DOCKER_PLATFORMS" \
+        --tag "$CI_REGISTRY_IMAGE:$CI_PIPELINE_IID" \
+        .
   tags:
     - gitlab-org-docker
 
+build:yoctocompat:
+  image: golang:1.17.13-bullseye
```

And add a comment here
```suggestion
  # Keep golang version aligned with latest yocto release
  image: golang:1.17.13-bullseye
```